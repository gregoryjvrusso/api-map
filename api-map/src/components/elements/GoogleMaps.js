import React, { useState, useEffect } from 'react'
import { Map, Marker, GoogleApiWrapper } from 'google-maps-react'

const mapStyles = {
    width: '90%',
    height: '70%'
};

const GoogleMaps = ({ latMap, lngMap }) => {
    return (
        <Map
            google={window.google}
            zoom={18}
            style={mapStyles}
            initialCenter={{ lat: latMap, lng: lngMap }}
            center={{ lat: latMap, lng: lngMap }}
        >
            <Marker
                name={'SOMA'}
                position={{ lat: latMap, lng: lngMap }} />
        </Map>
    )
}

export default GoogleApiWrapper({
    apiKey: 'AIzaSyAtxEFG6FZcvaVN-Ng6IE3_zUNHkhNsBWU'
})(GoogleMaps)