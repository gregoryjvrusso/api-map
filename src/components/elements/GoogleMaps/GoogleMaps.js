import React from 'react'
import { Map, Marker, GoogleApiWrapper } from 'google-maps-react'

const GoogleMaps = ({ latMap, lngMap }) => {
  return (
    <Map
      google={window.google}
      zoom={18}
      style={{ width: '100%', height: '50%', position: 'relative'}}
      initialCenter={{ lat: latMap, lng: lngMap }}
      center={{ lat: latMap, lng: lngMap }}
    >
      <Marker
        name={'SOMA'}
        position={{ lat: latMap, lng: lngMap }} />
    </Map>
  )
}

export default GoogleApiWrapper({
  apiKey: 'AIzaSyAtxEFG6FZcvaVN-Ng6IE3_zUNHkhNsBWU'
})(GoogleMaps)