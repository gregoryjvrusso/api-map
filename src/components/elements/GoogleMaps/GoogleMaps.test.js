import React from 'react'
import renderer from 'react-test-renderer';
import GoogleMaps from './GoogleMaps.js'

describe('GoogleMaps component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
    .create(<GoogleMaps latMap='-24.0252206' lngMap='-46.4931331,17' />)
    .toJSON();
    expect(render).toMatchSnapshot();
  })
})
