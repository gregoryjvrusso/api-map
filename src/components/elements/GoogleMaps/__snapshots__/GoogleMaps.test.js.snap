// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMaps component deve conseguir renderizar [snapshot] 1`] = `
<div>
  Loading...
</div>
`;
